glib/src/variant.rs

```diff
@@ -391,12 +391,46 @@ impl Variant {
     }
 
     /// Create an iterator over items in the variant.
+    ///
+    /// Note that this heap allocates a variant for each element,
+    /// which can be particularly expensive for large arrays.
     pub fn iter(&self) -> VariantIter {
         assert!(self.is_container());
 
         VariantIter::new(self.clone())
     }
 
+    /// Create an iterator over borrowed strings from a GVariant of type `as` (array of string).
+    ///
+    /// This will fail if the variant is not an array of with
+    /// the expected child type.
+    ///
+    /// A benefit of this API over [`Self::iter()`] is that it
+    /// minimizes allocation, and provides strongly typed access.
+    ///
+    /// ```
+    /// # use glib::prelude::*;
+    /// let strs = &["foo", "bar"];
+    /// let strs_variant: glib::Variant = strs.to_variant();
+    /// for s in strs_variant.array_iter_str()? {
+    ///     println!("{}", s);
+    /// }
+    /// # Ok::<(), Box<dyn std::error::Error>>(())
+    /// ```
+    pub fn array_iter_str<'a>(&'a self) -> Result<VariantStrIter<'a>, VariantTypeMismatchError> {
```

Comment:
We would probably want the same around `Variant::bytes()`?

Reply:
```
error: explicit lifetimes given in parameter types where they could be elided (or replaced with `'_` if needed by type declaration)
   --> glib/src/variant.rs:420:5
    |
420 |     pub fn array_iter_str<'a>(&'a self) -> Result<VariantStrIter<'a>, VariantTypeMismatchError> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D clippy::needless-lifetimes` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
```

Reply:
> We would probably want the same around Variant::bytes()?

For `aay` you mean?  I've used this type signature occasionally but it seems pretty rare.  It's also not very expensive to do `v.iter().map(|v| v.bytes())` right?

Reply:
That won't work because iteration creates a new variant, and the lifetime of the byte slice is constrained by the lifetime of that new variant.

But yes, I meant `aay`

Reply:
Right, you'd need to do instead something like:

```
let aay = &[b""].to_variant();
for buf in aay.iter() {
  let buf: &[u8] = buf.bytes();
  ...
}
```
which I think would be fine right?